### PR TITLE
Feature/cbs vp c api

### PR DESCRIPTION
**Pull request by IceTDrinker**
### Checklist 

(Use '[x]' to check the checkboxes, or submit the PR and then click the checkboxes)

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] The PR description links to the related issue (to link an issue, use '#XXX'.)
* [ ] The tests on AWS have been launched and are successful (comment with @slab-ci cpu_test and/or @slab-ci gpu_test to trigger the tests)
* [ ] The draft release description has been updated
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by IceTDrinker** [comment]
@slab-ci cpu_test

**Comment by IceTDrinker** [comment]
@slab-ci cpu_test

File: concrete-core-ffi/src/backends/default/engines/lwe_circuit_bootstrap_private_functional_packing_keyswitch_keys_generation.rs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+use concrete_core::prelude::*;
+use std::os::raw::c_int;
+
+/// Generate a new `LweCircuitBootstrapPrivateFunctionalPackingKeyswitchKeys64`.
+///
+/// Refer to `concrete-core` implementation for detailed documentation.
+///
+/// This function is [checked](crate#safety-checked-and-unchecked-functions).
+#[no_mangle]
+pub unsafe extern "C" fn default_engine_generate_new_lwe_circuit_bootstrap_private_functional_packing_keyswtich_keys_u64(
```
Comment (agnesLeroy) [inline concrete-core-ffi/src/backends/default/engines/lwe_circuit_bootstrap_private_functional_packing_keyswitch_keys_generation.rs:14]: typo in keyswitch

File: concrete-core-ffi/src/backends/default/engines/lwe_circuit_bootstrap_private_functional_packing_keyswitch_keys_generation.rs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                .unwrap(),
+        );
+        *result = Box::into_raw(heap_allocated_cbs_pfpksk);
+    })
+}
+
+/// [Unchecked](crate#safety-checked-and-unchecked-functions) version of
+/// [`default_engine_generate_new_lwe_circuit_bootstrap_private_functional_packing_keyswtich_keys_u64`]
+#[no_mangle]
+pub unsafe extern "C" fn default_engine_generate_new_lwe_circuit_bootstrap_private_functional_packing_keyswtich_keys_unchecked_u64(
```
Comment (agnesLeroy) [inline concrete-core-ffi/src/backends/default/engines/lwe_circuit_bootstrap_private_functional_packing_keyswitch_keys_generation.rs:54]: typo in keyswitch

File: concrete-core-ffi/src/backends/fftw/engines/lwe_ciphertext_discarding_bit_extraction.rs [hunk 3]
```diff
@@ -146,8 +145,8 @@ pub unsafe extern "C" fn fftw_engine_lwe_ciphertext_discarding_bit_extraction_un
         let fftw_engine = &mut (*fftw_engine);
         let default_engine = &mut (*default_engine);
 
-        let bootstrap_key = get_ref_checked(bootstrap_key).unwrap();
-        let keyswitch_key = get_ref_checked(keyswitch_key).unwrap();
+        let bootstrap_key = &(*bootstrap_key);
+        let keyswitch_key = &(*keyswitch_key);
```
Comment (agnesLeroy) [inline concrete-core-ffi/src/backends/fftw/engines/lwe_ciphertext_discarding_bit_extraction.rs:149]: :scream: thanks for the fix!

File: concrete-core-ffi/src/backends/default/engines/lwe_ciphertext_vector_decryption.rs [hunk 4]
```diff
@@ -124,8 +136,11 @@ pub unsafe extern "C" fn default_engine_decrypt_lwe_ciphertext_vector_unchecked_
...
             engine.decrypt_lwe_ciphertext_vector_unchecked(secret_key, &input);
-        *result = *engine
-            .retrieve_plaintext_vector_unchecked(&decrypted_plaintext_vector)
-            .as_mut_ptr();
+
+        let decrypted_plaintext_vector_as_vec =
+            engine.retrieve_plaintext_vector_unchecked(&decrypted_plaintext_vector);
+
+        let result_as_mut_slice = std::slice::from_raw_parts_mut(result, lwe_count);
+        result_as_mut_slice.copy_from_slice(&decrypted_plaintext_vector_as_vec);
```
Comment (agnesLeroy) [inline concrete-core-ffi/src/backends/default/engines/lwe_ciphertext_vector_decryption.rs:144]: Yeah alright, that was really stupid of me :sweat_smile:

**Review by agnesLeroy — APPROVED** [review]
Hello! You're doing it @IceTDrinker :muscle:! I just spotted a typo in a name otherwise it looks good to me :slightly_smiling_face:

**Pull request merged**